If HAVE_SETUPTERM isn't defined (because we didn't find setupterm, because we
linked with libtermcap instead of libcurses), then we define our own version,
but we still include <term.h>, so the prototypes conflict.  So just use curses.
--- krb5-1.4/src/appl/telnet/configure.in	2005-03-15 16:42:35.882145050 -0500
+++ krb5-1.4/src/appl/telnet/configure.in	2005-03-15 16:42:30.802808371 -0500
@@ -50,11 +50,12 @@
 #endif
 ])
 dnl
-AC_CHECK_LIB(termcap,main,AC_DEFINE(TERMCAP)
-LIBS="$LIBS -ltermcap",
+dnl AC_CHECK_LIB(termcap,main,AC_DEFINE(TERMCAP)
+dnl LIBS="$LIBS -ltermcap",
   AC_CHECK_LIB(curses,setupterm,LIBS="$LIBS -lcurses",
   AC_CHECK_LIB(ncurses,setupterm,LIBS="$LIBS -lncurses")
-))
+)
+dnl ))
 KRB5_AC_INET6
 AC_CHECK_FUNCS(setupterm)
 AC_CHECK_HEADER(termios.h,AC_DEFINE(USE_TERMIO) ac_termio=1)
@@ -96,11 +97,12 @@
 dnl from old telnetd/configure.in
 dnl
 dnl AC_PROG_INSTALL
-AC_CHECK_LIB(termcap,main,AC_DEFINE(TERMCAP)
-LIBS="$LIBS -ltermcap",
+dnl AC_CHECK_LIB(termcap,main,AC_DEFINE(TERMCAP)
+dnl LIBS="$LIBS -ltermcap",
   AC_CHECK_LIB(curses,setupterm,LIBS="$LIBS -lcurses",
   AC_CHECK_LIB(ncurses,setupterm,LIBS="$LIBS -lncurses")
-))
+)
+dnl ))
 dnl ... whole termios.h/termio.h/NO_CC_T thing again ...
 AC_HEADER_TIME
 dnl KRB5_AC_INET6
